analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -129,25 +130,78 @@ private void NextPosition(bool checkEOI = true)
             }
             else
             {
-                do
-                {
-                    line++;
-                }
-                while (line < lines.Count && lines[line].Length == 0);
-                position = 0;
-                if (checkEOI && !CanReadCurrentChar)
-                {
-                    throw new JsonException("Unexpected EOI" + AtLocationSuffix);
-                }
+                SeekToNextLine(checkEOI);
+            }
+        }
+
+        private void NthPosition(int n, bool checkEOI = true)
+        {
+            for (var i = 0; i < n; i++)
+            {
+                NextPosition(checkEOI);
+            }
+        }
+
+        private void SeekToNextLine(bool checkEOI)
+        {
+            do
+            {
+                line++;
+            }
+            while (line < lines.Count && lines[line].Length == 0);
+            position = 0;
+            if (checkEOI && !CanReadCurrentChar)
+            {
+                throw new JsonException("Unexpected EOI" + AtLocationSuffix);
             }
         }
 
         private void SkipWhiteSpace()
         {
-            while (CanReadCurrentChar && char.IsWhiteSpace(CurrentChar))
+            var isInMultiLineComment = false;
+            while (CanReadCurrentChar)
             {
-                NextPosition(false);
+                if (IsWhitespace())
+                {
+                    NextPosition(false);
+                }
+                else if (IsSingleLineComment())
+                {
+                    // We just need to read starting from the next line
+                    SeekToNextLine(false);
+                }
+                else if (IsCloseMultiLineComment())
+                {
+                    isInMultiLineComment = false;
+                    // need to skip */ part of the comment
+                    NthPosition(MultiLineCommentTokenLength, false);
```

Comment:
Is this legal?
```
*
/
```
If yes, let's add UT.
If no, we can do just `position += MultiLineCommentTokenLength` instead and remove `NthPosition` as that one does too many things.

We could also do this instead, for simplicity (remove the constant and `NthPosition`) - I find it more direct and easier to understand:
```
NextPosition(false); // Skip *
NextPosition(false); // Skip /
```

Reply:
No this is not legal. Actually, comments are not allowed in all JSON, we use it to support highlighting of our issues within JSON files, that makes it easy.

Reply:
Will do the suggested NextPosition way. It was like this before, but it looked smelly. But since you find it explanatory, I will leave it there